chia/full_node/mempool_manager.py

```diff
@@ -748,22 +773,23 @@ async def local_get_coin_records(names: Collection[bytes32]) -> List[CoinRecord]
         potential_txs.update(self._conflict_cache.drain())
         txs_added = []
         for item in potential_txs.values():
-            cost, status, error = await self.add_spend_bundle(
+            info = await self.add_spend_bundle(
                 item.spend_bundle,
                 item.npc_result,
                 item.spend_bundle_name,
                 item.height_added_to_mempool,
                 self.get_coin_records,
             )
-            if status == MempoolInclusionStatus.SUCCESS:
-                txs_added.append((item.spend_bundle, item.npc_result, item.spend_bundle_name))
+            if info.status == MempoolInclusionStatus.SUCCESS:
+                txs_added.append(NewPeakItem(item.spend_bundle_name, item.spend_bundle, item.npc_result))
+            mempool_item_removals.extend(info.removals)
         log.info(
             f"Size of mempool: {self.mempool.size()} spends, "
             f"cost: {self.mempool.total_mempool_cost()} "
             f"minimum fee rate (in FPC) to get in for 5M cost tx: {self.mempool.get_min_fee_rate(5000000)}"
         )
         self.mempool.fee_estimator.new_block(FeeBlockInfo(new_peak.height, included_items))
-        return txs_added
+        return NewPeakInfo(txs_added, mempool_item_removals)
```

Comment:
I think you should have unit tests to ensure you get the correct removals returned here and in the other plaes if `MempoolManager`